nocasedict/_nocasedict.py

```diff
@@ -234,7 +226,7 @@ def __getitem__(self, key):
           TypeError: Key does not have a ``lower()`` method.
           KeyError: Key does not exist (case-insensitively).
         """
-        k = self._real_key(key)
+        k = _real_key(key)
```

Comment:
Just a question. Was there a reason you moved _real_key outside the class???

Reply:
It did not use self anymore after removing its access to `self.allow_unnamed_keys`, so the choice were between making it a static method or moving it out.